src/snowflake/connector/s3_storage_client.py

```diff
@@ -441,25 +529,23 @@ def download_chunk(self, chunk_id: int) -> None:
                 _range = f"{chunk_id * chunk_size}-"
 
             response = self._send_request_with_authentication_and_retry(
-                url,
-                "GET",
-                _resource,
-                chunk_id,
+                url=url,
+                verb="GET",
+                retry_id=chunk_id,
                 headers={"Range": f"bytes={_range}"},
             )
             if response.status_code in (200, 206):
                 self.write_downloaded_chunk(chunk_id, response.content)
             response.raise_for_status()
 
     def transfer_accelerate_config(self) -> bool:
-        url = self.endpoint + "/?accelerate"
-        _resource = self._construct_canonicalized_element(
-            bucket_name=self.s3location.bucket_name, subresource={"accelerate": None}
-        )
+        query_parts = (("accelerate", ""),)
```

Comment:
I found that we used to construct these URLs, but then I needed to extract some parts of them later on, query parameters were the ones that were the most complicated ones to parse out.

So instead I switched to having them in the form of `Tuple[Tuple[str, str], ...]`, where the strings refer to each key and value.
This way I don't need to parse them out of the URL and they are easy to turn into a string like `accelerate` in this case, but it also makes me not having to parse the URL again later on.